### PR TITLE
Task/173-transaction-item AND Task/174-transaction-list

### PR DESCRIPTION
**Pull request by TomasPetrik**

### CODE REVIEW

File: lib/extensions/date_extensions.dart [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import 'package:intl/intl.dart';
+
+extension DateParse on DateTime {
```
Comment (storkandstars) [inline lib/extensions/date_extensions.dart:3]: Please, move this extension to `view/extensions` folder as it is related to view layer only

File: lib/view/widgets/transaction_item.dart [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import 'package:flutter/material.dart';
+import 'package:gap/gap.dart';
+import 'package:simplio_app/data/model/transaction.dart';
+import 'package:simplio_app/extensions/date_extensions.dart';
+import 'package:simplio_app/extensions/number_extensions.dart';
```
Comment (storkandstars) [inline lib/view/widgets/transaction_item.dart:5]: This extension should be inside `view/extensions` folder

File: lib/view/widgets/transaction_item.dart [hunk 3]
```diff
@@ -0,0 +1,121 @@
+import 'package:flutter/material.dart';
+import 'package:gap/gap.dart';
+import 'package:simplio_app/data/model/transaction.dart';
+import 'package:simplio_app/extensions/date_extensions.dart';
```
Comment (storkandstars) [inline lib/view/widgets/transaction_item.dart:4]: This extension should be inside `view/extensions` folder

File: pubspec.yaml [hunk 4]
```diff
@@ -42,6 +42,7 @@ dependencies:
   gap: ^2.0.0
   mobile_scanner: ^2.0.0
   url_launcher: ^6.1.5
+  infinite_scroll_pagination: ^3.2.0
```
Comment (storkandstars) [inline pubspec.yaml:45]: Why do we use another package for pagination? We are able to achieve this easily inside flutter.

**Review by storkandstars — CHANGES_REQUESTED** [review]

File: lib/extensions/date_extensions.dart (hunk 1 above)
Comment (TomasPetrik) [reply lib/extensions/date_extensions.dart:3]: Done.

File: lib/view/widgets/transaction_item.dart (hunk 3 above)
Comment (TomasPetrik) [reply lib/view/widgets/transaction_item.dart:4]: Yes. It was merged but not with proper file location. I will fix it.

File: lib/view/widgets/transaction_item.dart (hunk 2 above)
Comment (TomasPetrik) [reply lib/view/widgets/transaction_item.dart]: I moved it there.

File: lib/logic/cubit/transactions/prices/transactions_cubit.dart [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:infinite_scroll_pagination/infinite_scroll_pagination.dart';
+import 'package:simplio_app/data/model/transaction.dart';
+import 'package:simplio_app/data/repositories/transaction_repository.dart';
+import 'package:simplio_app/view/themes/constants.dart';
+
+part 'transactions_state.dart';
+
+class TransactionsCubit extends Cubit<TransactionsState> {
```
Comment (storkandstars) [inline lib/logic/cubit/transactions/prices/transactions_cubit.dart:10]: Why is this file into sub-directory of transactions? `logic/cubit/transactions/prices`

**Review by storkandstars — CHANGES_REQUESTED** [review]

File: lib/logic/cubit/transactions/prices/transactions_cubit.dart (hunk 5 above)
Comment (TomasPetrik) [reply lib/logic/cubit/transactions/prices/transactions_cubit.dart:10]: Moved it to  logic/cubit/transactions/

**Review by jvrastil — CHANGES_REQUESTED** [review]
There is an error when going to `Transactions` tan and then to any other tab:
```
[VERBOSE-2:ui_dart_state.cc(198)] Unhandled Exception: Bad state: Cannot emit new states after calling close
#0      BlocBase.emit (package:bloc/src/bloc_base.dart:96:9)
#1      TransactionsCubit.loadTransactions (package:simplio_app/logic/cubit/transactions/transactions_cubit.dart:45:7)
<asynchronous suspension>
```

**Comment by TomasPetrik** [comment]
> 

@jvrastil 

You can check it. It's fixed now.

**Review by jvrastil — APPROVED** [review]
looks good now

**Review by storkandstars — APPROVED** [review]

**Pull request merged**